Bio/SeqIO/GckIO.py

```diff
@@ -71,11 +71,15 @@ def GckIterator(handle):
     Note that a GCK file can only contain one sequence, so this
     iterator will always return a single record.
     """
+    # check if file is empty
+    if not handle.read(1):
+        raise ValueError("Empty file.")
+
     # Skip file header
     # GCK files start with a 24-bytes header. Bytes 4 and 8 seem to
     # always be 12, maybe this could act as a magic cookie. Bytes
     # 17-20 and 21-24 contain variable values of unknown meaning.
-    _read(handle, 24)
+    _read(handle, 23)
```

Comment:
Performance is not going to be an issue, but I think doing a single 24 bytes read is better:

```python
temp = handle.read(24)
if not temp:
    raise ValueError("Empty file.")
elif len(temp) < 24:
    raise ValueError("Cannot read 24 bytes from handle")
    # This preserves the current exception, perhaps should be more specific?
    # raise ValueError("Truncated file? 24 byte header incomplete.")
del temp
```

Reply:
See also the ``_read_header()`` helper function you defined in one of the other files... might be worth sharing that?